set.mm

```diff
@@ -788160,6 +789266,1078 @@ Ring homomorphisms (extension)
+      UWAUURUWDMVRUNZVGZUWTUXOLUWAMVTVCZUURUWTVCZUWDUWTVCZUXSUWTVCAUXTUVBUVTAMQ
+      XHWCUVSUYAUVTAUVBUYAALUWTUURAUWTLUXBXJZXKXLWQAUVTUYBUVBAUVLCHIJLMNOPQRSTX
+      MXNZUWTMUXRUURUWDUXAUXRVSZWOWNUWALHXOUNVCZUVBUWDLVCUXOUXSUOAUYFUVBUVTALHX
+      RUNVCUYFAHLNOAHLXPVGMVQPAUXEMVQVCQMXSWBXQXTLHYAWBWCZAUVBUVTYGUWAUWDUWTLUY
+      DAUWTLUOUVBUVTUXBWCYBLHMUWEUXRUURUWDUXGPUYEYCWNALUWTUOUVBUVTUYCWCYDYHUWAU
+      XPUYFUVTUWMUXLUXNVKUXQUYGUWRUXJCGLHUWEUVLUWHRUXGUAYEYFYIUWAUWJUURAUVBUVTU
+      WJUURUOAUURCUVLHIJLMNOPQRSTYJYKXJXBWQYLYMYNYOYPYQYRWTYSUVFUUNUVJUGCUVFUUL
+      CVCZVAUUKUUSUUMUVIUUTUVCAUYHYTAUYHUUMUVIUOUVDABUULCDEFGHIJKLMNOPQRSTUAUBU
+      CUDUEUUFUUAUUBUUCYIYNUUDWBUUGUUEUGUFCUUJKUUHUUI $.
+
+    $( ` F ` is linear regarding the multiplication.  (Contributed by AV,
+       28-Feb-2025.) $)
+    rngqiprnglin $p |- ( ph -> A. a e. B A. b e. B
+                  ( F ` ( a .x. b ) ) = ( ( F ` a ) ( .r ` P ) ( F ` b ) ) ) $=
+      ( cv co cfv cmulr wceq wcel wa cec cop cvv crg cbs eqid cqus ovexi adantr
+      crng simpl quseccl0 syl2an rngqiprngghmlem1 sylan2 simpr rngqiprnglinlem3
+      a1i ringcld xpsmul rngqiprnglinlem2 c2idl ressmulr oveqd rngqiprnglinlem1
+      eqcomd syl eqtrd opeq12d eqtr2d anim1i 3anass sylibr rngqiprngimfv syldan
+      w3a rngcl oveq12d 3eqtr4d ralrimivva ) ANUHZOUHZIUIZKUJZWOKUJZWPKUJZEUKUJ
+      ZUIZULNOCCAWOCUMZWPCUMZUNZUNZWQGUOZJWQIUIZUPZWOGUOZJWOIUIZUPZWPGUOZJWPIUI
+      ZUPZXAUIZWRXBXFXPXJXMFUKUJZUIZXKXNMUKUJZUIZUPXIXFXJXKXMXNFMXAEXQXSUQURFUS
+      UJZMUSUJZUFYAUTZYBUTZFUQUMXFFHGVAUDVBVLAMURUMXESVCZAHVDUMZXCXJYAUMXEPXCXD
+      VEZYACGLHFVDWOUCUDTYCVFVGXEAXCXKYBUMYGAWOCHIJLMPQRSTUAUBVHVIZAYFXDXMYAUMX
+      EPXCXDVJZYACGLHFVDWPUCUDTYCVFVGXEAXDXNYBUMYIAWPCHIJLMPQRSTUAUBVHVIZAWOCWP
+      FGHIJLMPQRSTUAUBUCUDVKXFYBMXSXKXNYDXSUTZYEYHYJVMXQUTYKXAUTVNXFXRXGXTXHXFX
+      GXRAWOCWPFGHIJLMPQRSTUAUBUCUDVOVTXFXTXKXNIUIXHXFXSIXKXNXFIXSXFLHVPUJZUMZI
+      XSULAYMXEQVCLHMIYLRUAVQWAVTVRAWOCWPHIJLMPQRSTUAUBVSWBWCWDAXEWQCUMZWRXIULX
+      FYFXCXDWJZYNXFYFXEUNYOAYFXEPWEYFXCXDWFWGCHIWOWPTUAWKWAABWQCDEFGHIJKLMPQRS
+      TUAUBUCUDUEUFUGWHWIXFWSXLWTXOXAXEAXCWSXLULYGABWOCDEFGHIJKLMPQRSTUAUBUCUDU
+      EUFUGWHVIXEAXDWTXOULYIABWPCDEFGHIJKLMPQRSTUAUBUCUDUEUFUGWHVIWLWMWN $.
+
+    $d B a b $.  $d F a b $.  $d P a b $.  $d R a b $.
+    $( ` F ` is a homomorphism of non-unital rings.  (Contributed by AV,
+       21-Feb-2025.) $)
+    rngqiprngho $p |- ( ph -> F e. ( R RngHomo P ) ) $=
+      ( va vb crng wcel cghm co cv cmulr wceq wral crngh rngqiprng rngqiprngghm
+      cfv wa rngqiprnglin jca eqid isrnghm syl21anbrc ) AHUHUIEUHUIKHEUJUKUIZUF
+      ULZUGULZIUKKUSVGKUSVHKUSEUMUSZUKUNUGCUOUFCUOZUTKHEUPUKUINACDEFGHIJLMNOPQR
+      STUAUBUCUDUQAVFVJABCDEFGHIJKLMNOPQRSTUAUBUCUDUEURABCDEFGHIJKLMUFUGNOPQRST
+      UAUBUCUDUEVAVBUFUGCHEIKVIRSVIVCVDVE $.
+
+    $( ` F ` is an isomorphism of non-unital rings.  (Contributed by AV,
+       21-Feb-2025.) $)
+    rngqiprngim $p |- ( ph -> F e. ( R RngIsom P ) ) $=
```

Comment:
hypothesis C can be removed using eqid

Reply:
As said already above, I think the flow of proof towards a specific goal justifies that all these theorems are within one block, and therefore it happens that some hypotheses are not necessary for some of these theorems.